resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/config/ClientConfiguration.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.palisade.service.resource.config;
+
+import com.netflix.discovery.EurekaClient;
+import com.netflix.discovery.shared.Application;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.netflix.eureka.EurekaServiceInstance;
+
+import uk.gov.gchq.palisade.Generated;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class ClientConfiguration {
+    private Map<String, URI> client;
+
+    @Autowired(required = false)
+    private EurekaClient eurekaClient;
```

Comment:
@Autowired should never appear in production code (tests are fine). Replace with constructor injection. You will also need a no-op.

Reply:
Since this is present in other services and has been there for a while, I'll raise a bug for the non-resource-service usages and fix just this instance

Reply:
Expand that bug... none of this discovery code should be in the services. The services should be dumb, remember this works on an inversion of control pattern. If you want to check connectivity it should be external, like the launcher for example.

Reply:
I want all this gone.